pygsheets/spreadsheet.py

```diff
@@ -60,10 +60,19 @@ def sheet1(self):
 
     @property
     def named_ranges(self):
-        """All named ranges in thi spreadsheet"""
+        """All named ranges in this spreadsheet"""
         return [DataRange(namedjson=x, name=x['name'], worksheet=self.worksheet('id', x['range'].get('sheetId', 0)))
                 for x in self._named_ranges]
 
+    @property
+    def protected_ranges(self):
+        """All protected ranges in this spreadsheet"""
+        request = self.client.service.spreadsheets().get(spreadsheetId=self.id, fields="sheets/(properties/sheetId,protectedRanges)", includeGridData=True)
+        response = self.client._execute_request(self.id, request, False)
+        return [DataRange(protectedjson=x, worksheet=self.worksheet('id', sheet['properties']['sheetId']))
```

Comment:
need to set protected=True as DataRange  param

Reply:
`protected` will be `True` if `protect_id is not None` (see `DataSheet.protected` getter), so `protectedjson` is quite enough to satisfy this requirement, isn't it?

Reply:
Ohh my bad. I didn't check the getter, i just checked if protected is working as intended. Yup this would satisfy the requirement. But i think you missed a `not` in getter.

Reply:
Definitely must be `is not None`, [added fix](https://github.com/nithinmurali/pygsheets/pull/175/commits/cba40533048071b335d395ea28818176f526aed7) and [removed](https://github.com/nithinmurali/pygsheets/pull/175/commits/cb89440c3a06339bdf8f00f89249a5e6a7dbb1eb) redundant `_protected` attribute.